modules/civicrm_entity_actions/civicrm_entity_actions_action_schedule.inc

```diff
@@ -41,26 +41,26 @@ function civicrm_entity_actions_action_schedule_action_info() {
 
 function civicrm_entity_actions_action_schedule_add_action($entity, $context = array()) {
   $base_url = str_replace('_', '-', $context['entity_type']); // civicrm-action_schedule
-  drupal_goto($base_url . '/add');
+  rules_action_drupal_goto($base_url . '/add');
```

Comment:
If you use this function then this module will need to add a dependency to Rules in the .info file

Reply:
alternatively you could add a if (module_exists('rules')) { new function, else drupal_goto